### DISCUSSION TITLE
Maybe expose the setScroll api?

### DISCUSSION THREAD

**Discussion by leyiang** [katspaugh/wavesurfer.js#3659] [Q&A] [answered]
Hi maintainer, thank you for your amazing work. I really love it.

Currently I'm developing an app that build on top of your project.
And I'm trying to add a "drag timeline" feature. (I know there's an autoScroll option, but my feature needs a little different)
In the documentation, I can find two APIs related: "getScroll" and "setScrollTime".
getScroll returns scrollLeft value, and setScrollTime take time as parameter.

I find this to be a little inconvenient, because I need to getScroll (in px), and calculate drag dx (in px), and get the total width (in px), then convert it into percentage, then getDuration( in seconds), then get the time to be set.

My workaround for now:
```ts
function setScrollPx(x: number) {
    const scrollEL = el.firstElementChild?.shadowRoot?.querySelector(".scroll");

    if( scrollEL ) {
        scrollEL.scrollLeft = x;
    }
}
```
I checked the source code of getScrollTime api, which essentailly called [the privte method setScroll of renderer](https://github.com/katspaugh/wavesurfer.js/blob/3f10c5a0b745068ef8e2317cabc90a9263d117a9/src/renderer.ts#L264)

My point: why don't we just expose this api? I'm currently fine with my workaround. But maybe it could be helpful for others?

**Comment by katspaugh** [comment, marked as answer]
Sure, it would be nice to expose a setScroll method.

**Comment by leyiang** [comment]
#3660
